subprojects/core/src/integTest/groovy/org/gradle/api/tasks/DeferredTaskDefinitionIntegrationTest.groovy

```diff
@@ -313,6 +313,32 @@ class DeferredTaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         failure.assertHasCause("Task with name 'task1' not found")
     }
 
+    def "delegate of TaskProvider.configure is set properly"() {
```

Comment:
I think this would be better mixed into the specific "can configure a task" scenarios, as it seems like we're just doubling up on scenarios here. Either that, or we should add corresponding separate "delegate is set correctly" test cases for all of the action taking methods, e.g. createLater().